### PR TITLE
HAWQ-1099. Output yaml file should not contain Bucketnum attribute with random distributed table.

### PR DESCRIPTION
**Pull request by xunzhang**

### CODE REVIEW

**Comment by xunzhang** [comment]
cc @wcl14 @ictmalili

File: tools/bin/hawqregister [hunk 1]
```diff
@@ -60,14 +60,18 @@ def option_parser():
...
             sys.exit(1)
-    if D['Bucketnum'] <= 0:
-        logger.error('Bucketnum should not be zero, please check your yaml configuration file.')
-        sys.exit(1)
+    if D['Distribution_Policy'].startswith('DISTRIBUTED BY'):
+        if D.get('Bucketnum') is None:
+            logger.error('Wrong configuration yaml file format: "%s" attribute does not exist.\n See example in "hawq register --help".' % attr)
+            sys.exit(1)
+        if D['Bucketnum'] <= 0:
+            logger.error('Bucketnum should not be zero, please check your yaml configuration file.')
```
Comment (ictmalili) [inline tools/bin/hawqregister]: Could we modify the error message.  "Should be a positive integer"

File: tools/bin/hawqregister [hunk 2]
```diff
@@ -432,7 +436,8 @@ class HawqRegister(object):
         if len(params[Format_FileLocations]['Files']):
             files, sizes = [params['DFS_URL'] + d['path'] for d in params[Format_FileLocations]['Files']], [d['size'] for d in params[Format_FileLocations]['Files']]
         encoding = params['Encoding']
-        self._set_yml_dataa(Format, files, sizes, params['TableName'], params['%s_Schema' % Format], params['Distribution_Policy'], params[Format_FileLocations], params['Bucketnum'], partitionby,\
+        bucketNum = params['Bucketnum'] if params['Distribution_Policy'].startswith('DISTRIBUTED BY') else 6
```
Comment (ictmalili) [inline tools/bin/hawqregister]: I don't think the else should be "6", we need get the default bucket number from HAWQ (GUC: default_hash_table_bucket_number), or we don't need assign bucketnum here if we want to create table, i.e, don't specify in creating table DDL.

**Comment by xunzhang** [comment]
@ictmalili done, please review again.

**Comment by xunzhang** [comment]
Merged into master, thanks.

**Pull request closed without merging**